Lib9c/Action/ClaimItems.cs

```diff
@@ -0,0 +1,107 @@
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [ActionType(ActionTypeText)]
+    public class ClaimItems : GameAction, IClaimItems
+    {
+        private const string ActionTypeText = "claim_items";
+
+        public List<(Address address, List<FungibleAssetValue> fungibleAssetValues)> ClaimData { get; private set; }
+
+        public ClaimItems()
+        {
+        }
+
+        public ClaimItems(List<(Address, List<FungibleAssetValue>)> claimData)
+        {
+            ClaimData = claimData;
+        }
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            ImmutableDictionary<string, IValue>.Empty
+                .Add(ClaimDataKey, ClaimData.Select(x =>
+                {
+                    var serializedFungibleAsetValues = x.fungibleAssetValues.Select(x => x.Serialize()).Serialize();
+
+                    return (x.address, serialized: serializedFungibleAsetValues);
+                }).Serialize());
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            ClaimData = plainValue[ClaimDataKey].ToStateList()
+                .Select((tuple =>
+                {
+                    return (tuple.Item1, tuple.Item2.ToList((x => x.ToFungibleAssetValue())));
+                })).ToList();
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            context.UseGas(1);
+
+            var states = context.PreviousState;
+            var itemSheet = states.GetSheets(containItemSheet: true).GetItemSheet();
+
+            foreach (var (avatarAddress, fungibleAssetValues) in ClaimData)
+            {
+                var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
+                var inventory = states.GetInventory(inventoryAddress)
+                                ?? throw new FailedLoadStateException(
+                                    ActionTypeText,
+                                    GetSignerAndOtherAddressesHex(context, inventoryAddress),
+                                    typeof(Inventory),
+                                    inventoryAddress);
+
+                foreach (var fungibleAssetValue in fungibleAssetValues)
+                {
+                    if (fungibleAssetValue.Currency.DecimalPlaces != 0)
+                    {
+                        throw new ArgumentException(
+                            $"DecimalPlaces of fungibleAssetValue for claimItems are not 0: {fungibleAssetValue.Currency.Ticker}");
+                    }
+
+                    var parsedTicker = fungibleAssetValue.Currency.Ticker.Split("_");
+                    if (parsedTicker.Length != 3
+                        || parsedTicker[0] != "Item"
+                        || (parsedTicker[1] != "NT" && parsedTicker[1] != "T")
+                        || !int.TryParse(parsedTicker[2], out var itemId))
```

Comment:
I'll try to define a new type like `ItemTokenCurrency: Curreny` that validates `Currency.Ticker` when constructing instances, in working hours.

Reply:
I think I can handle this. Thank you for suggestion!

Reply:
***C# STRUCT CAN'T INHERIT STRUCT***